database/migrations/2022_10_18_103120_create_osu_user_performance_rank_highest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class CreateOsuUserPerformanceRankHighest extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('osu_user_performance_rank_highest', function (Blueprint $table) {
+            $table->charset = 'utf8mb4';
+            $table->collation = 'utf8mb4_general_ci';
+
+            $table->unsignedInteger('user_id');
+            $table->tinyInteger('mode');
+            $table->integer('rank')->default(0);
+            $table->timestamp('updated_at')->useCurrent()->useCurrentOnUpdate();
+
+            $table->primary(['user_id', 'mode']);
+        });
+        DB::statement('
+            ALTER TABLE `osu_user_performance_rank_highest` PARTITION BY RANGE (`mode`) (
+                PARTITION p0 VALUES LESS THAN (1),
+                PARTITION p1 VALUES LESS THAN (2),
+                PARTITION p2 VALUES LESS THAN (3),
+                PARTITION p3 VALUES LESS THAN (4)
+            );
```

Comment:
@peppy are these the right partitions? I copied from another table, but they weren't listed in the schema u sent

Reply:
```sql
root@db-master.osu.io:osu> show create table osu.osu_user_performance_rank_highest\G;
***************************[ 1. row ]***************************
Table        | osu_user_performance_rank_highest
Create Table | CREATE TABLE `osu_user_performance_rank_highest` (
  `user_id` int unsigned NOT NULL,
  `mode` tinyint NOT NULL,
  `rank` int NOT NULL DEFAULT '0',
  `updated_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`user_id`,`mode`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci ROW_FORMAT=DYNAMIC
/*!50100 PARTITION BY RANGE (`mode`)
(PARTITION p0 VALUES LESS THAN (1) ENGINE = InnoDB,
 PARTITION p1 VALUES LESS THAN (2) ENGINE = InnoDB,
 PARTITION p2 VALUES LESS THAN (3) ENGINE = InnoDB,
 PARTITION p3 VALUES LESS THAN (4) ENGINE = InnoDB) */
```